src/core/or/connection_edge.c

```diff
@@ -863,18 +863,31 @@ export_hs_client_circuit_id(edge_connection_t *edge_conn,
   if (protocol != HS_CIRCUIT_ID_PROTOCOL_HAPROXY)
     return;
 
+  const origin_circuit_t *circ;
   char *buf = NULL;
   const char dst_ipv6[] = "::1";
   /* See RFC4193 regarding fc00::/7 */
-  const char src_ipv6_prefix[] = "fc00:dead:beef:4dad:";
+  const char src_ipv6_prefix[] = "fc00:";
   uint16_t dst_port = 0;
   uint16_t src_port = 1; /* default value */
   uint32_t gid = 0; /* default value */
+  uint32_t rp_addr = 0;
+  uint16_t rp_port = 0;
 
   /* Generate a GID and source port for this client */
   if (edge_conn->on_circuit != NULL) {
-    gid = TO_ORIGIN_CIRCUIT(edge_conn->on_circuit)->global_identifier;
+    circ = CONST_TO_ORIGIN_CIRCUIT(edge_conn->on_circuit);
+    gid = circ->global_identifier;
     src_port = gid & 0x0000ffff;
+    if (get_options()->HiddenServiceExportRendPoint) {
+      extend_info_t *extend_info = circ->build_state->chosen_exit;
+      if (BUG(extend_info == NULL)) {
+        tor_free(buf);
```

Comment:
I think you should replace this `tor_free(buf)` here with a `goto` statement to the end of the function where `tor_free(buf)` is called as well.

Reply:
Fixed in `1609fe9`.